### PR TITLE
Quota integration test was not closing launched controllers

### PR DESCRIPTION
**Pull request by derekwaynecarr**
This helps debug https://github.com/kubernetes/kubernetes/issues/25037 where the controller was stuck in a hot loop trying to address a quota that no longer existed because the controller itself was never stopped.

/cc @deads2k

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Marking WIP until I verify it with a few runs.

**Comment by derekwaynecarr** [comment]
hmm, must be missing something, closing for now.

**Comment by derekwaynecarr** [comment]
So this fixes the log spam in the reported issue, but the test is still subject to timing out it looks like.

File: test/integration/quota_test.go [hunk 1]
```diff
@@ -194,6 +197,7 @@ func scale(t *testing.T, namespace string, clientset *clientset.Clientset) {
 
 		switch cast := event.Object.(type) {
 		case *api.ReplicationController:
+			fmt.Printf("Found %v of %v replicas\n", int(cast.Status.Replicas), target)
```
Comment (deads2k) [inline test/integration/quota_test.go]: `t.Logf`?

**Comment by deads2k** [comment]
nit.  lgtm.

File: test/integration/quota_test.go (hunk 1 above)
Comment (derekwaynecarr) [inline test/integration/quota_test.go]: this didnt appear to print in a panic.

**Comment by derekwaynecarr** [comment]
@k8s-bot - test this issue https://github.com/kubernetes/kubernetes/issues/23365

**Comment by derekwaynecarr** [comment]
@k8s-bot - test this issue #23365

**Comment by fabioy** [comment]
Manually merging, since it may help with tests.

**Pull request merged**